climada_petals/engine/warn.py

```diff
@@ -0,0 +1,238 @@
+        Possible values: 'DILATION', 'EROSION', 'MEDIANFILTERING'
+    sizes : list
+        Size of kernel of every operation given.
+    gradual_decr : bool
+        Used to define whether the highest warn levels should be gradually decreased (if True)
+        to the lowest level (e.g., level 3, 2, 1, 0)
+        or larger steps are allowed (e.g., from warn level 5 directly to 1)
+    change_sm : bool
+        If True, the levels of too small regions are changed to its surrounding levels
+    size_sm : int
+        Defining what too small regions are. Number of coordinates
+    """
+    # Explanation of defaults (also maybe transfer to config, talk to Emanuel)
+    OPERATIONS = ['DILATION', 'EROSION', 'DILATION', 'MEDIANFILTERING']
+    SIZES = [2, 3, 7, 15]
+    GRADUAL_DECREASE = True
+    CHANGE_SMALL_REGIONS = True
+    SIZE_TOO_SMALL = 150
+
+    def __init__(self, warn_levels, operations=OPERATIONS, sizes=SIZES,
+                 gradual_decr=GRADUAL_DECREASE, change_sm=CHANGE_SMALL_REGIONS, size_sm=SIZE_TOO_SMALL):
+        """Initialize FilterData."""
+        self.warn_levels = warn_levels
+        self.gradual_decr = gradual_decr
+        if len(operations) != len(sizes):
+            LOGGER.warning('For every operation a filter size is needed and the other way around. '
+                           'Please input the same number of operations and filter sizes.')
+        if not all(item in ['DILATION', 'EROSION', 'MEDIANFILTERING'] for item in operations):
+            raise ValueError("The operation is not defined. "
+                           "Please select 'EROSION', 'DILATION', or 'MEDIANFILTERING'.")
+
+        self.operations = operations
+        self.sizes = sizes
+        self.change_sm = change_sm
+        self.size_sm = size_sm
+
+
+class Warn:
+    """Warn definition. Generate a warning, i.e., 2D map of points with assigned warn levels. Operations, their order,
+    and their influence (kernel sizes) can be selected to fit the users purpose. Further properties can be chosen which
+    define the warning generation. The functionality of reducing heterogeneity in a map can be applied to different
+    inputs, e.g. MeteoSwiss windstorms, TCs, impacts, etc.
+
+    Attributes
+    ----------
+    warning : np.array
+        Warning generated by class. Warn level for every point of input map.
+    coord : np.array
+        Coordinates of warning map.
+    warn_levels : list
+        Warn levels that define the bins in which the input_map will be classified in.
+    """
+
+    # operations that can be applied in filtering algorithm
+    filtering_operations = {
+        "DILATION": lambda bin_m, size: skimage.morphology.dilation(bin_m, skimage.morphology.disk(size)),
+        'EROSION': lambda bin_m, size: skimage.morphology.erosion(bin_m, skimage.morphology.disk(size)),
+        'MEDIANFILTERING': lambda bin_m, size: skimage.filters.median(bin_m, np.ones((size, size)))
```

Comment:
Now that I think of it, maybe instead of using `lambda` methods you could define proper for each operation functions.